vital/types/local_cartesian.cxx

```diff
@@ -0,0 +1,446 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief This file contains the implementation of a local geographical offset coordinate system.
+ */
+
+#include <vital/types/local_cartesian.h>
+#include <vital/types/geodesy.h>
+
+#include <iomanip>
+#include <stdexcept>
+
+namespace kwiver {
+namespace vital {
+
+const double PI = 3.14159265358979323e0;
+const double PI_OVER_2 = (PI / 2.0e0);
+const double TWO_PI = (2.0 * PI);
+const double DEG_TO_RAD = (PI / 180.0);
+const double RAD_TO_DEG = (180.0 / PI);
+const double AD_C = 1.0026000;            /* Toms region 1 constant */
+const double COS_67P5 = 0.38268343236508977;  /* cosine of 67.5 degrees */
+
+/**
+ * @Brief Local Cartesian Conversion Math Utility
+ *
+ * Base on the NGA GeoTrans library
+ * https://earth-info.nga.mil/GandG/update/index.php?action=home
+ *
+ * This class is a cleaned up version of the LocalCartesian and Geocentric classes provided in GeoTrans
+ * This allows a user to define a local cartesian coordinate system with any origin (expressed in WGS84)
+ */
+class local_cartesian::geotrans
+{
+public:
+  geotrans(geo_point const& origin, double orientation)
+  {
+    // Using WGS84 ellipsoid
+    semiMajorAxis = 6378137.0;
+    inv_f = 298.257223563;
+    flattening = 1 / inv_f;
+
+    Geocent_e2 = 2 * flattening - flattening * flattening;
+    Geocent_ep2 = (1 / (1 - Geocent_e2)) - 1;
+    set_origin(origin, orientation);
+  }
+
+  /**
+   * @brief Set origin of the cartesian system as a geo_point
+   *
+   * Set local origin parameters as inputs and sets the corresponding state variables.
+   * NOTE : If the origin changes, this method needs to be called again to recompute
+   *        variables needed in the conversion math.
+   *
+   * @param [in] geo_point       : Geograpical origin of the cartesian system
+   * @param [in] orientation     : Orientation angle of the local cartesian coordinate system,
+   *                               in radians along the Z axis, normal to the earth surface
+   */
+  void set_origin(geo_point const& origin, double orientation)
+  {
+    if (origin.is_empty())
+    {
+      throw std::runtime_error("Origin geo_point is empty");
+    }
+
+    double N0;
+    double val;
+
+    auto loc = origin.location(kwiver::vital::SRID::lat_lon_WGS84);
+
+    LocalCart_Origin_Lat = loc[1] * DEG_TO_RAD;
+    LocalCart_Origin_Long = loc[0] * DEG_TO_RAD;
+    if (LocalCart_Origin_Long > PI)
```

Comment:
Missing braces. (Also, you might want to rewrite this in terms of `fmod` instead...)